packages/dialtone-vue2/components/filter_pill/filter_pill.vue

```diff
@@ -0,0 +1,203 @@
+        <dt-button
+          data-qa="dt-filter-pill__button"
+          icon-position="right"
+          importance="outlined"
+          kind="muted"
+          :size="size"
+          :aria-busy="loading"
+          :aria-label="loading ? label : undefined"
+          :class="{ 'd-filter-pill--active': active }"
+          :disabled="disabled"
+          @click="isPopoverOpen = true"
+        >
+          <span v-if="!loading">
+            {{ label }}
+          </span>
+          <dt-skeleton
+            v-else
+            :text-option="{ width: loadingSkeletonWidth }"
+          />
+          <template #icon="{ iconSize }">
+            <dt-icon-chevron-down
+              data-qa="dt-filter-pill__icon"
+              class="d-filter-pill__icon"
+              :size="iconSize"
+            />
+          </template>
+        </dt-button>
+        <dt-button
+          v-if="showReset"
+          data-qa="dt-filter-pill__reset-button"
+          class="d-filter-pill__reset"
+          circle
+          importance="clear"
+          kind="muted"
+          :size="size"
+          :aria-label="resetButtonAriaLabel"
+          @click="$emit('reset', $event)"
+        >
+          <template #icon="{ iconSize }">
+            <dt-icon-close :size="iconSize" />
+          </template>
+        </dt-button>
+      </div>
+    </template>
+    <template #headerContent>
+      <!-- @slot Header content of the opened popover -->
+      <slot name="headerContent" />
+    </template>
+    <template #content>
+      <!-- @slot Main content of the opened popover -->
+      <slot name="content" />
+    </template>
+    <template #footerContent>
+      <!-- @slot Footer content of the opened popover -->
+      <slot name="footerContent" />
+    </template>
+  </dt-popover>
+</template>
+
+<script>
+import { DtPopover } from '@/components/popover';
+import { DtSkeleton } from '@/components/skeleton';
+import { BUTTON_SIZE_MODIFIERS, DtButton } from '@/components/button';
+import { DtIconChevronDown, DtIconClose } from '@dialpad/dialtone-icons/vue2';
+
+export default {
+  name: 'DtFilterPill',
+
+  components: {
+    DtPopover,
+    DtButton,
+    DtIconClose,
+    DtIconChevronDown,
+    DtSkeleton,
+  },
+
+  inheritAttrs: false,
+
+  props: {
+    /**
+     * Sets the button highlighted styling
+     */
+    active: {
+      type: Boolean,
+      default: false,
+    },
+
+    /**
+     * HTML disabled attribute
+     */
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+
+    /**
+     * Label of the button
+     */
+    label: {
+      type: String,
+      default: undefined,
+    },
+
+    /**
+     * Determines if the filter is loading
+     */
+    loading: {
+      type: Boolean,
+      default: false,
+    },
+
+    /**
+     * Sets the size of the skeleton
+     * accepts CSS width attribute values
+     */
+    loadingSkeletonWidth: {
+      type: String,
+      default: '128px',
+    },
```

Comment:
Changing this changes the entire width of the button rather than the width of the skeleton. Intended?

Reply:
As the skeleton is growing/shrinking it'd be expected that the button size change, right? The use-case I saw for this was setting the loading-skeleton-width to the actual text-size which could be calculated either on the client or I can add that here but I'd be more flexible if I left the client to decide the width of the loading skeleton.

Reply:
If the client is receiving the text from a server it won't know its size until the request is complete in which case it is done loading. Calculating width based on the length of a string even if we had it is VERY difficult due to variable width fonts, i18n, and text scaling. Basically the same problem we had in this ticket where we were trying to dynamically change the size of the field based on the language and it was impossible to size it correctly.

https://dialpad.atlassian.net/browse/DP-107950

Reply:
Ok, setting it to the text size was too ambitious 😅 but having the flexibility to change the size is ok, right?